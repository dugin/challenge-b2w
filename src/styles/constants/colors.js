export default {

    whiteSmoke: '#f7f7f7',
    pattensBlue: '#dbe2e8',
    azure : '#eafbff',
    brightTurquoise: '#06eac3',
    caribbeanGreen: '#05d2af',
    baliHai: '#7D97AD',
    shadow: 'rgba(46,61,73,.2)',
    sapphire: '#00175B',
}