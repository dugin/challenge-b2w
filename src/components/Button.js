import React from 'react';
import styled from 'styled-components';
import colors from "../styles/constants/colors";
import PropTypes from 'prop-types';

const Wrapper = styled.button`
      box-shadow: 5px 5px 25px 0 ${colors.shadow};
      border-radius: 5px;
      border: none;
      background-color: ${props => props.disabled ? colors.pattensBlue : colors.brightTurquoise  };
      pointer-events: ${props => props.disabled ? 'none' : 'auto'  };
      color: ${colors.sapphire};
      padding: 14px 48px;
      font-size: 16px;
      outline: none;
      
      &:hover{
      box-shadow:2px 4px 8px 0 ${colors.shadow};
      background-color: ${colors.caribbeanGreen};
      cursor: pointer;
      }
`;

const Button = (props) => {

    return (
        <Wrapper disabled={props.isLoading} onClick={() => props.onPress()}>
            {props.isLoading ? 'Loading... ' : props.text }
        </Wrapper>
    )
};

Button.propTypes = {
    text: PropTypes.string.isRequired,
    onPress: PropTypes.func.isRequired,
    isLoading: PropTypes.bool
};

export default Button;