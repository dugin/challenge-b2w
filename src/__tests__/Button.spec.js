import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow, mount} from 'enzyme';
import Button, {Wrapper} from "../components/Button";
import colors from "../styles/constants/colors";

describe('Button Component', () => {

    const button = shallow(<Button isLoading={true} onPress={jest.fn()} text={'Next'}/>);
    const wrapper = button.find(Wrapper);

    it('renders properly', () => {
        expect(toJson(button)).toMatchSnapshot();
    });

    it('should be disabled', () => {
        expect(wrapper).toHaveStyleRule('background-color', colors.pattensBlue);
        expect(wrapper).toHaveStyleRule('pointer-events', 'none');
    });

    it('should contain the text "Loading..."', () => {
        expect(wrapper.childAt(0).text()).toEqual('Loading... ');
    });

    it('click should trigger properly', () => {
        const button = mount(<Button onPress={jest.fn()} text={'Next'}/>);

        button.simulate('click');

        expect(button.props().onPress).toHaveBeenCalled();
    })
});