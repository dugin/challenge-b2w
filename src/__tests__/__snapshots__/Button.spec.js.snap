// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders properly 1`] = `
<styled.button
  disabled={true}
  onClick={[Function]}
>
  Loading... 
</styled.button>
`;
